lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -0,0 +1,141 @@
+define((require, exports, module) ->
+  'use strict'
+  oop = require('../../lib/oop')
+  TextHighlightRules = require('../../mode/text_highlight_rules').TextHighlightRules
+
+  PapeeriaLatexHighlightRules = ->
+    ###*
+    * We maintain a stack of nested LaTeX semantic types (e.g. "document", "section", "list"
+    * to be able to provide context for autocompletion and other functions. 
+    * Stack is constructed by the background highlighter;
+    * its elements are then propagated to * the editor session and become 
```

Comment:
I thought I asked to explain _how_ are they propagated. It is not quite obvious how `bgTokenizer.getState` becomes aware of all this stuff